api/proto/teleport/legacy/types/types.proto

```diff
@@ -5737,17 +5754,32 @@ message ServerInfoV1 {
   ];
 }
 
-// ServerInfoSpecV1 contains fields used to match Nodes to this ServerInfo.
-message ServerInfoSpecV1 {
+// ServerResourceMatcher is a set of properties that is used to match on resources.
+message ServerResourceMatcher {
   // AWSInfo contains attributes to match to an EC2 instance.
   message AWSInfo {
     // AccountID is an AWS account ID.
     string AccountID = 1 [(gogoproto.jsontag) = "account_id"];
     // InstanceID is an EC2 instance ID.
     string InstanceID = 2 [(gogoproto.jsontag) = "instance_id"];
   }
+
+  // Labels is the set of labels to match to a Server.
+  wrappers.LabelValues Labels = 1 [
+    (gogoproto.jsontag) = "labels",
+    (gogoproto.customtype) = "Labels"
+  ];
+
   // AWS matches an EC2 instance.
-  AWSInfo AWS = 1 [(gogoproto.jsontag) = "aws,omitempty"];
+  AWSInfo AWS = 2 [(gogoproto.jsontag) = "aws,omitempty"];
+}
+
+// ServerInfoSpecV1 contains fields used to match Nodes to this ServerInfo.
+message ServerInfoSpecV1 {
+  reserved 1;
+  reserved "AWS";
+  // ResourceMatchers is a list of resource matchers.
+  repeated ServerResourceMatcher ResourceMatchers = 2 [(gogoproto.jsontag) = "resources"];
```

Comment:
Are you wanting to explicitly version `ServerResourceMatcher` like we do for other resources?

Reply:
I think that since we decided to just do 1-1 server-serverInfo matching for now, I'll take the other matching code out and save this discussion for a future RFD.